### DISCUSSION TITLE
Desktop Access with Active Directory

### DISCUSSION THREAD

**Discussion by saied-1990** [gravitational/teleport#27060] [Q&A] [answered]
Hi Team , 

I'm getting the below issues after configure integration with Active Directory

RDP connection failed: access to windows_desktop denied. User does not have permissions. Confirm Windows user.
<img width="1438" alt="Screenshot 2023-05-27 at 4 20 40 PM" src="https://github.com/gravitational/teleport/assets/101193037/c7ae50b1-e777-4068-8bcc-bafd72911987">

**Comment by webvictim** [comment, marked as answer]
This is an issue with the permissions in your Teleport cluster.

You should have a role with a set of permissions a bit like this:

```yaml
spec:
  allow:
    windows_desktop_logins:
    - Administrator
    - Gus
    windows_desktop_labels:
      '*': '*'
```

Then when you go to connect to a Windows desktop, you should see a list of valid usernames you can use for the given labels in your role:

![image](https://github.com/gravitational/teleport/assets/897821/d642794a-6c46-4fd6-a6bb-9b44b98e4c3c)

If you don't see the available usernames in the drop-down menu, your permissions are not applying correctly. You will need to fix this before desktop access will work.

If you do see the usernames in the drop-down and still get this error, log out of the Teleport web UI and log back in again to make sure your permissions are up-to-date.
